nh_observations/observations.py

```diff
@@ -45,14 +45,36 @@ class NhClinicalPatientObservation(orm.AbstractModel):
 
     @classmethod
     def get_description(cls, append_observation=True):
+        """
+        Return a label for the observation suitable for display.
+
+        :param append_observation:
+        :return:
+        :rtype: str
+        """
         description = \
             super(NhClinicalPatientObservation, cls).get_description()
         if append_observation:
             description += ' Observation'
         return description
 
     def get_obs_field_order(self):
-        return self._required
+        """
+        Get the intended order of fields which can be used for various things
+        such as display order in the mobile observation form (achieved by
+        ordering the field descriptions in the form description).
+
+        :return: List of field names.
+        :rtype: list
+        """
+        if not hasattr(self, 'fields_order'):
+            raise NotImplementedError(
+                "Tried to get the order of fields for model '{model_name}' "
+                "but could not find a `fields_order` class attribute. Either "
+                "one must be added or the `get_obs_field_order()` method "
+                "should be overridden."
+            )
+        return self.fields_order
```

Comment:
Has this method `get_obs_field_order` changed just to use the `fields_order` that seems to be newly set for therapeutics? I can see you had to add this for `nh_neurological/models/nh_clinical_patient_observation_neurological.py`, yet didn't delete the `_required` declaration that was originally used in that model. 
Was there an issue with the implementation of Therapeutic obs that wouldn't allow you to use `_required` as the other obs have? Otherwise, wouldn't you need to also add this `fields_order` to `nh_observations/models/nh_clinical_patient_observation_height.py`, `nh_blood_glucose/models/nh_clinical_patient_observation_blood_glucose.py`,  etc? (I tried to look for ews but couldn't find the model. Of course the height and bg only have 1 field, so you might not actually need an 'order' for a single field, but wouldn't this raise an error because the model doens't have it defined when this triggers?)
If the Therapeutics can work with `_required` instead, why was this changed (legit question - I don't know :) )

Reply:
Yeah you're right `_required` couldn't be used for therapeutic so I had to introduce the separate `_field_order` attribute. That's because the `_required` field is also used for somewhere else for other purposes (I think it's when deciding if an obs is partial or not). It was okay to use it for both purposes before because only one observation used this method and all its fields were required which meant that a full ordering could also be specified by its `_required`. With therapeutic none of the fields are `_required` so the field order needs to be separate to avoid messing up other validation.

The other obs are okay because they don't use this method at all because they define their form description explicitly rather than using the newer, more 'automatic' way of creating the form description that relies on this method. I want to eventually change all the obs to use the newer way of generating form descriptions (which involves calling this method) so that they don't each have to specify their own.

Does that make sense?

Reply:
It makes sense :+1: , but wouldn't we want to add that as tech debt and then do it for all the obs in one go? Or would the Neurological fail? 
I take Neuro shouldn't be using this method as you mentioned (and if I got it right), so just wondering if we want to add unused code at this point in time and not do it all at a later date (if the application will still function as expected of course)

Reply:
I am taking the following as 'not 100% sure'. We developed it and tested it, but haven't used it since, so not sure how it would work atm.
When we developed F&F, we had only a couple of actual 'mandatory' fields. Fields you had to submit in order for the observation to go through. F&F though didn't have Partials, so it's quite similar if not the same in that aspect as Therapeutic would. How could the order work for F&F but not for Therapeutics? https://github.com/NeovaHealth/openeobs/blob/master/nh_food_and_fluid/models/food_and_fluid.py

Reply:
Sorry I've probably not been very clear. A newer, more 'automatic' way of generating the form description was first introduced when Neurological was created. Neurological was the first obs to use that functionality, but it was only the first iteration of it, just enough to get Neurological working with it. So the above method is used in `develop` when getting the form description for Neurological.

I've now come along and created a new observation and would like to use the still immature but newer way of generating the form observations. That's involved building on it a little bit so it can accommodate the newer observation which.

F&F doesn't use the newer way of generating a form description which is why it doesn't need to satisfy the above method that is only used with that functionality. In the model there is just a form description dictionary explicitly defined.

Reply:
:+1: